src/SideNav.js

```diff
@@ -8,31 +8,47 @@ class SideNav extends Component {
     this.id = props.id || `sidenav_${idgen()}`;
   }
 
+  show () {
+    $(this._trigger).sideNav('show');
+  }
+
+  hide () {
+    $(this._trigger).sideNav('hide');
+  }
+
+  destroy () {
+    $(this._trigger).sideNav('destroy');
+  }
+
   componentDidMount () {
     const { options = {} } = this.props;
     $(this._trigger).sideNav(options);
   }
 
   render () {
-    const { children, ...props } = this.props;
+    const { children, fixed, ...props } = this.props;
     delete props.id;
     delete props.trigger;
     delete props.options;
+    delete props.fixed;
+    delete props.showOnLarge;
     return (
       <span>
         { this.renderTrigger() }
-        <ul id={this.id} className='side-nav' {...props}>
+        <ul id={this.id} className={'side-nav' + (fixed ? ' fixed' : '')} {...props}>
           {children}
         </ul>
       </span>
     );
   }
 
   renderTrigger () {
-    const { trigger } = this.props;
+    const { trigger, showOnLarge } = this.props;
+    // delete this.props.showOnLarge;
     return React.cloneElement(trigger, {
       ref: (t) => (this._trigger = `[data-activates=${this.id}]`),
-      'data-activates': this.id
+      'data-activates': this.id,
+      className: ((trigger.props.className !== undefined ? trigger.props.className : '') + (showOnLarge ? ' show-on-large' : ''))
```

Comment:
Since this is quite hard to understand, how about we add `classnames` package and use here instead? see [here](https://github.com/react-materialize/react-materialize/blob/ebfd5c9f6bf89c7bb8e7390a83246c9217ec33ca/src/Card.js#L7) for an example